src/env.cc

```diff
@@ -1068,6 +1068,13 @@ Environment::~Environment() {
   }
 
   delete external_memory_accounter_;
+  if (cpu_profiler_) {
+    for (auto it : pending_profiles_) {
```

Comment:
```suggestion
    for (auto& it : pending_profiles_) {
```

Otherwise this copies every entry?